presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -345,6 +351,50 @@ protected RelationType visitShowColumns(ShowColumns showColumns, AnalysisContext
     }
 
     @Override
+    protected RelationType visitShowCreateTable(ShowCreateTable showCreateTable, AnalysisContext context)
+    {
+        QualifiedObjectName tableName = createQualifiedObjectName(session, showCreateTable, showCreateTable.getName());
+
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+        if (!tableHandle.isPresent()) {
+            throw new SemanticException(MISSING_TABLE, showCreateTable, "Table '%s' does not exist", tableName);
+        }
+
+        ConnectorTableMetadata connectorTableMetadata = metadata.getTableMetadata(session, tableHandle.get()).getMetadata();
+
+        List<ColumnMetadata> columns = connectorTableMetadata.getColumns().stream().filter(column -> !column.isHidden()).collect(toImmutableList());
+        Map<String, Object> properties = connectorTableMetadata.getProperties();
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("CREATE TABLE ");
+        builder.append(formatName(tableName.getCatalogName())).append(".");
+        builder.append(formatName(tableName.getSchemaName())).append(".");
+        builder.append(formatName(tableName.getObjectName()));
+        builder.append(" (").append(NEW_LINE + "  ");
+        Joiner.on("," + NEW_LINE + "  ").appendTo(builder, transform(columns, element -> formatName(element.getName()) + " " + element.getType()));
+        builder.append(NEW_LINE).append(")");
+
+        if (!properties.isEmpty()) {
+            builder.append(NEW_LINE).append("WITH (").append(NEW_LINE).append("  ");
+            Joiner.on("," + NEW_LINE + "  ").appendTo(builder, transform(properties.entrySet(), entry -> entry.getKey() + " = " + entry.getValue()));
+            builder.append(NEW_LINE).append(")");
+        }
+
+        SingleColumn singleColumn = new SingleColumn(new StringLiteral(builder.toString()), Optional.of("Create Table"));
+        Query query = simpleQuery(new Select(false, ImmutableList.of(singleColumn)));
+
+        return process(query, context);
+    }
+
+    private static String formatName(String name)
+    {
+        if (CharMatcher.WHITESPACE.matchesAnyOf(name) || name.contains("'")) {
```

Comment:
This should be a regex that matches valid unquoted identifiers. Something like `[a-z_][a-z0-9_]`